### PR TITLE
add fchain validation for fee token observation

### PR DESCRIPTION
**Pull request by 0xAustinWang**
https://smartcontract-it.atlassian.net/browse/CCIP-3980

### CODE REVIEW

File: commit/tokenprice/validate_observation.go [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+	if err := validateObservedTokenPrices(obs.FeedTokenPrices); err != nil {
+		return fmt.Errorf("failed to validate observed token prices: %w", err)
+	}
+
+	return nil
+}
+
+func validateFChain(fChain map[cciptypes.ChainSelector]int) error {
+	for chainSelector, f := range fChain {
+		if f < 0 {
```
Comment (asoliman92) [inline commit/tokenprice/validate_observation.go:43]: I don't think we should allow 0 fChain.
```suggestion
		if f <= 0 {
```

File: commit/tokenprice/validate_observation_test.go [hunk 2]
```diff
@@ -64,3 +64,40 @@ func Test_validateObservedTokenPrices(t *testing.T) {
...
+			},
+			expErr: true,
+		},
+		{
+			name: "FChain valid",
+			fChain: map[cciptypes.ChainSelector]int{
+				12: 6,
+				7:  9,
+			},
+			expErr: false,
```
Comment (asoliman92) [inline commit/tokenprice/validate_observation_test.go:88]: Maybe add a test for zero fChain as well

File: commit/tokenprice/validate_observation.go (hunk 1 above)
Comment (asoliman92) [reply commit/tokenprice/validate_observation.go:43]: same for other validations in other files.

**Review by asoliman92 — APPROVED** [review]
